### PR TITLE
Add support for watermarks

### PR DESCRIPTION
**Pull request by ylatuya**

### CODE REVIEW

File: VAS.Core/Common/Watermark.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		/// The offset y, normalized from 0 to 1 based on output video height
+		/// </summary>
+		public double OffsetY;
+
+		/// <summary>
+		/// Gets the default watermark.
+		/// </summary>
+		/// <returns>The default watermark.</returns>
+		/// <param name="position">Position.</param>
+		public static Watermark GetDefaultWatermark (WatermarkPosition position, VideoStandard videoStandard)
```
Comment (vguzmanp) [inline VAS.Core/Common/Watermark.cs]: Missing tests

File: system_uuid/.deps/vas-system-uuid.Po [hunk 2]
```diff
@@ -0,0 +1 @@
+# dummy
```
Comment (vguzmanp) [inline system_uuid/.deps/vas-system-uuid.Po]: ?

File: VAS.Core/Common/Watermark.cs (hunk 1 above)
Comment (vguzmanp) [inline VAS.Core/Common/Watermark.cs]: I don't like this method's name. It's not getting the default watermark, it's configuring many things, calculating the offsets and creating the watermark.

File: VAS.UI.Gtk2/UI/Dialog/VideoEditionProperties.cs [hunk 3]
```diff
@@ -56,8 +56,14 @@ public VideoEditionProperties (Window parent)
 				OutputDir = mediafilechooser2.CurrentPath;
 			};
 
-			watermarkcheckbutton.Active = true;
-			watermarkcheckbutton.Sensitive = !App.Current.LicenseManager.LicenseStatus.Limited;
+			watermarkcheckbutton.Active = App.Current.Config.AddWatermark;
+			//FIXME: This Logic should be changed once Longomatch initializes LicenseManager
+			if (App.Current.LicenseManager != null) {
```
Comment (vguzmanp) [inline VAS.UI.Gtk2/UI/Dialog/VideoEditionProperties.cs]: If it is null, we should set it to true, or fail, shouldn't we?

File: VAS.Core/Common/Watermark.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+		/// <summary>
+		/// The offset x, normalized from 0 to 1 based on output video height
+		/// </summary>
+		public double OffsetX;
+
+		/// <summary>
+		/// The offset y, normalized from 0 to 1 based on output video height
+		/// </summary>
+		public double OffsetY;
```
Comment (vguzmanp) [inline VAS.Core/Common/Watermark.cs]: Make these properties

File: VAS.UI.Gtk2/UI/Dialog/VideoEditionProperties.cs [hunk 5]
```diff
@@ -125,6 +134,12 @@ protected virtual void OnButtonOkClicked (object sender, System.EventArgs e)
 			encSettings.EnableAudio = audiocheckbutton.Active;
 			encSettings.EnableTitle = descriptioncheckbutton.Active;
 
+			if (watermarkcheckbutton.Active) {
+				encSettings.Watermark = Watermark.GetDefaultWatermark (WatermarkPosition.TOP_RIGHT, encSettings.VideoStandard);
+			} else {
+				encSettings.Watermark = null;
+			}
```
Comment (vguzmanp) [inline VAS.UI.Gtk2/UI/Dialog/VideoEditionProperties.cs]: Why are all the settings not configured if there's not a watermark?

File: VAS.Core/StyleConf.cs [hunk 6]
```diff
@@ -241,6 +241,10 @@ public class StyleConf
 		public const string PrelightEye = "images/events/el_eye_p" + IMAGE_EXT;
 		public const string InsensitiveEye = "images/events/el_eye_i" + IMAGE_EXT;
 
+		//Watermark
+		public const int WatermarkPadding = 20;
+		public const double WatermarkHeightNormalization = 0.087;
```
Comment (vmartos) [inline VAS.Core/StyleConf.cs]: Is the Height of the Watermark normalized from 0 to 1, the story says that always it should be 8,7% from the total Height of the video

File: VAS.UI.Gtk2/UI/Dialog/VideoEditionProperties.cs (hunk 3 above)
Comment (vmartos) [inline VAS.UI.Gtk2/UI/Dialog/VideoEditionProperties.cs]: Longomatch can configure the watermark in preferences, that's why i do:

watermarkcheckbutton.Active = App.Current.Config.AddWatermark;

First. 

And I added a FIXME to change it in the future when LongoMatch should have always the LicenseManager initialized, if not it should throw.

File: VAS.Tests/Utils.cs [hunk 7]
```diff
@@ -266,6 +267,44 @@ public DummyDashboardManagerVM (DashboardVM vm)
...
+		}
+
+		public Image LoadImage (string name, int width = 0, int height = 0)
+		{
+			return GetDummyImage ();
+		}
+
+		public void Register (Assembly assembly)
+		{
+			throw new NotImplementedException ();
```
Comment (vguzmanp) [inline VAS.Tests/Utils.cs]: Do you really want to throw here?

File: VAS.Tests/Core/Common/TestWatermark.cs [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+using NUnit.Framework;
+using VAS.Core.Common;
+
+namespace VAS.Tests.Core.Common
+{
+	[TestFixture]
+	public class TestWatermark
+	{
+		[Test]
+		public void TestWatermarkPosition_TopLeft ()
```
Comment (vguzmanp) [inline VAS.Tests/Core/Common/TestWatermark.cs]: Tests don't need to start with `Test`

File: VAS.Tests/Core/Common/TestWatermark.cs [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+			Assert.AreEqual (expectedOffsetX, watermark.OffsetX);
+			Assert.AreEqual (expectedOffsetY, watermark.OffsetY);
+			Assert.AreEqual (expectedHeight, watermark.Image.Height);
+		}
+
+		[Test]
+		public void TestWatermarkPosition_BottomLeft ()
+		{
+			VideoStandard videoStandard = VideoStandards.P480;
```
Comment (vguzmanp) [inline VAS.Tests/Core/Common/TestWatermark.cs]: Why do you change the videoStandard in each test?

Comment (vmartos) [inline VAS.Tests/Core/Common/TestWatermark.cs]: Just to ensure that the offsets and image height is correctly calculated with different video sizes inside the ConfigureNewWatermark function.

Comment (vguzmanp) [inline VAS.Tests/Core/Common/TestWatermark.cs]: So... those are different tests, for the position, and for the standard

File: VAS.Tests/Core/Common/TestWatermark.cs [hunk 10]
```diff
@@ -25,7 +25,7 @@ namespace VAS.Tests.Core.Common
 	public class TestWatermark
 	{
 		[Test]
-		public void ConfigureWatermark_PositionTopLeft ()
+		public void ConfigureWatermark_PositionTopLeft_VideoStandard_P720 ()
```
Comment (vguzmanp) [inline VAS.Tests/Core/Common/TestWatermark.cs]: The third part of the test should be the expected result.

https://fluendo.atlassian.net/wiki/display/APP/Coding+Style#CodingStyle-Testnaming

File: VAS.Tests/Core/Common/TestWatermark.cs (hunk 9 above)
Comment (vguzmanp) [inline VAS.Tests/Core/Common/TestWatermark.cs]: You renamed the tests, but I meant that they should be different tests.

Ideally, all the combinations between videostandard & position. As that is quite redundant, at least set one of the two variables, and test with all the values of the other variable.

For example:
```
P480-Top_left
P480-Top_right
P480-Bot_left
P480-Bot_right

Top_left-P240
Top_left-P480
Top_left-P720
Top_left-P1080
```

**Pull request merged**